react/components/contexts/auth-context/auth-context.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { verify } from "yonius";
+import { API as RipeIdAPI } from "ripe-id-api";
+
+import { RipeContext } from "../ripe-context";
+
+export const AuthContext = React.createContext();
+
+export class AuthProvider extends Component {
+    static contextType = RipeContext;
+
+    static get propTypes() {
+        return {
+            options: PropTypes.shape({
+                clientId: PropTypes.string.isRequired,
+                clientSecret: PropTypes.string.isRequired,
+                redirectUrl: PropTypes.string.isRequired
+            })
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.ripeIdApi = new RipeIdAPI(this.props.options);
+        this.state = {
+            loaded: false,
+            logoutMessage: null,
+            account: null,
+            tokens: null,
+            ripeIdApi: this.ripeIdApi
+        };
+    }
+
+    async login(email, password) {
+        const response = await this.ripeIdApi.login(email, password);
+        verify(response.sid, "No session ID found", 404);
+        verify(response.username, "No username found", 404);
+        verify(response.tokens, "No tokens list found", 404);
+
+        this.ripeIdApi.sessionId = response.sid;
+        this.setState({
+            account: response.username,

```

Comment:
To make it similar to the oAuthContext, it would be nice to populate the account with the account object and not the username. Is it possible? Or at least call this username instead of account.

Reply:
The response to login has: `sid` a string, `username` a string too, and `tokens` a list of strings. Should I save it all under account? I think on the other context, account is not the same. I could call `ripeIdApi.selfAccount` and store it tho. Just an extra call but it would mean both contexts have the same data.

Reply:
Yap, that would be nice. Or remove the account and just add username